### ISSUE TITLE
[FEATURE] Approve or slow down node OS upgrade

### ISSUE THREAD

**Issue by try2codesecure** [harvester/harvester#3087]
**Is your feature request related to a problem?**
not exactly, but is upgrade related

**Describe alternatives you've considered**
none

**Additional context**

From an OPS pov it would be nice to be able to slow down or approve an node os upgrade on the following points:

![ts_upgrade_phases-7336371ca32df446a0f9da46e79729b6](https://user-images.githubusercontent.com/4202158/199470877-aaf0ad1a-7509-4a5d-b9d8-dd614d6ba92b.png)

Especially when your're using services with an application based clustering (MySQL, elastic, ...) you would like to check all your HA services on the upgraded node **before** the upgrade proceeds to the next node.

**Comment by yasker** [comment]
Interesting. @try2codesecure Do you have a more specific example of a workload that requires this kind of operation?

**Comment by try2codesecure** [comment]
Hi yasker. Sure.
Actually i'm operating an elastic & mysql HA cluster on my harvester cluster
(3x Harvester baremetal, 3x VM's mysql, 3x VM's elastic).
As of now these cluster are only used for various less important services, so i shutdown them before upgrade as recommended. In the future it would be nice when i can operate them while an upgrade in HA mode.

The mysql cluster runs in multi primary mode, full replication on all nodes (pxc). The elastic cluster shards it's data over all 3 nodes (1 primary, 1 replica on any node per shard). Both of them can tolerate 1 gone node to stay operable. Each of the VM's has 1 bounded longhorn volume with no replica (only stored on host).

* Now the first harvester hypervisor goes down (& the VM's).
The MySQL cluster stays operable, only one of three node goes down.
The elastic cluster starts to reshard the gone replicas on the remaining node & restores the missing primary shards from the existing replicas (still operable).

* Then the 1st harvester HV comes back up after the OS upgrade & start all services. 
Actually the upgrade process waits for the HV node to became healthy. But at this point the VM's aren't fully started yet & the services doesn't finished (or even started) the cluster sync.

* So the harvester upgrade continues to the 2nd node (with the needed running VM's with the current cluster state from the services) shut down the VM'S & upgrade the HV. These services now gone down with no chance to recover from the previous downtime of the 1st node.

* At the latest by the third node the services are down and needs a hand to recover.

**Comment by Martin-Weiss** [comment]
Why do we shutdown VMs instead of doing a migration?
In case we migrate them we should not see downtime from the application side of things..

**Comment by try2codesecure** [comment]
Good point, but i can't migrate those VM's out because they are too big to be stored on the remaining nodes.
That's why I don't have replicas from these VM's.
Furthermore i can store more data when i shard the data (e.g. in elastic).
